.github/workflows/linux_tests.yaml

```diff
@@ -167,7 +167,7 @@ jobs:
               -o ${{ runner.temp }}/artifacts/functional_lcov.info
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
```

Comment:
Apparently Codecov requires a token?
https://github.com/codecov/codecov-action/issues/1274

Reply:
Someone needs to set up a secret (I have access to the key in Codedov, but not this repo).
https://docs.codecov.com/docs/adding-the-codecov-token#github-actions

Reply:
Honestly, we should just strip out the codecov checks, they haven't worked since Raphael left the team.

Reply:
Let me know if I should remove these steps, or merge this as is ;-)

Reply:
Merge it as-is. I can follow up with another PR to strip out the codecov stuff.